src/CountryProvider.php

```diff
@@ -92,7 +91,11 @@ private function findCountryCode(Request $request): string
         }
 
         try {
-            return $this->reader->country($request->getClientIp())->country->isoCode ?: $this->fallbackCountry;
+            if (null === $this->reader && !empty($_SERVER['GEOIP2_DATABASE'])) {
+                $this->reader = new Reader($_SERVER['GEOIP2_DATABASE']);
+            }
+
+            return ($this->reader)()->country($request->getClientIp())->country->isoCode ?: $this->fallbackCountry;
```

Comment:
does it really make sense to instantiate the callback every time? Or shouldn't we rather store the callback result?